pravega/flink-connectors
# [Issue-181] Implement Flink TableFactory

Signed-off-by: Vijay Srinivasaraghavan vijayaraghavan.srinivasaraghavan@emc.com

**Change log description**
  * Provides Pravega Table factory for source and sink
  * Provides Pravega connector descriptor to programmatically invoke Table API/SQL
  * Supports YAML Pravega configurations to access Pravega streams using SQL client interface 
  * deprecated `FlinkPravegaJsonTableSink` and `FlinkPravegaJsonTableSource` implementations

**Purpose of the change**
This PR address the issues https://github.com/pravega/flink-connectors/issues/181 and https://github.com/pravega/flink-connectors/issues/182

**What the code does**
It provides Pravega specific implementations for [table factory](https://github.com/apache/flink/blob/master/flink-libraries/flink-table-common/src/main/java/org/apache/flink/table/factories/TableFactory.java) and [connector descriptor](https://github.com/apache/flink/blob/master/flink-libraries/flink-table-common/src/main/java/org/apache/flink/table/descriptors/ConnectorDescriptor.java) which allows accessing Pravega streams as Table to perform both source and sink operations. It also supports accessing Pravega streams through SQL client interface.  

**How to verify it**
- `./gradlew clean build` should pass
- Use SQL client shell and validate the table API access (details provide in documentation which is part of PR)
